gusto/time_discretisation/explicit_runge_kutta.py

```diff
@@ -114,45 +143,85 @@ def setup(self, equation, apply_bcs=True, *active_labels):
         """
         super().setup(equation, apply_bcs, *active_labels)
 
-        if not self.increment_form:
-            self.field_i = [Function(self.fs) for i in range(self.nStages+1)]
+        if self.rk_formulation == RungeKuttaFormulation.predictor:
+            self.field_i = [Function(self.fs) for _ in range(self.nStages+1)]
+        elif self.rk_formulation == RungeKuttaFormulation.increment:
+            self.k = [Function(self.fs) for _ in range(self.nStages)]
+        elif self.rk_formulation == RungeKuttaFormulation.linear:
+            self.field_rhs = Function(self.fs)
         else:
-            self.k = [Function(self.fs) for i in range(self.nStages)]
+            raise NotImplementedError(
+                'Runge-Kutta formulation is not implemented'
+            )
 
     @cached_property
     def solver(self):
-        if self.increment_form:
+        if self.rk_formulation == RungeKuttaFormulation.increment:
             return super().solver
-        else:
+
+        elif self.rk_formulation == RungeKuttaFormulation.predictor:
             # In this case, don't set snes_type to ksp only, as we do want the
-            # outer Newton iteration
+            # outer Newton iteration. This is achieved by not calling the
+            # "super" method, in which the default snes_type is set to ksp_only
```

Comment:
I'm not sure we need a nonlinear solve here? ksp only should be sufficient.

Reply:
Probably a very good point. I think we've decided to resolve this in #560